management/management_error.go

```diff
@@ -21,13 +21,25 @@ type managementError struct {
 	Message    string `json:"message"`
 }
 
-func newError(r io.Reader) error {
-	m := &managementError{}
-	if err := json.NewDecoder(r).Decode(m); err != nil {
-		return err
+func newError(response *http.Response) error {
```

Comment:
Quite a few improvements to this func:

- Preserve status code from the response instead of relying purely that the body contains the status code as well
- Preserve status code when we fail to decode the response when this is not a valid JSON